src/util/validationUtils.ts

```diff
@@ -13,3 +13,26 @@ export function validateSearchParams(query: RequestQuery) {
     throw new BadRequestError(`Parameters ${invalidParams.join(', ')} are not valid for search`);
   }
 }
+
+/*
+ * Gathers parameters from both the query and the FHIR parameter request body resource
+ */
+export function gatherParams(query: RequestQuery, parameters?: fhir4.Parameters) {
+  const gatheredParams: Record<string, any> = { ...query };
```

Comment:
Thoughts on replacing the `any`? I think the correct typing here would be `string | boolean | undefined | number | string[]` which is a mouthful 😅

Reply:
Hmmm- this is a tricky one. Currently, `gatherParams` is meant to handle anything that we'd throw at it with `$package`, but we're pretty likely to expand on the kinds of params that we'd gather in the future. When I look at the valueX options in `ParametersParameter` type, this list does cover most of the common ones, but there are plenty of other types that could come in as parameters on some operation in the future. I would lean toward keeping it `any` as it could capture all of that weird stuff as well as needed (and this function's stated purpose is to collect anything we need from a query). But I could be convinced otherwise if we think the `any` here mitigates enough other risk.